awx/ui_next/src/components/Lookup/CredentialsLookup.jsx

```diff
@@ -0,0 +1,165 @@
+import React from 'react';
+import PropTypes from 'prop-types';
+import { withI18n } from '@lingui/react';
+import { withRouter } from 'react-router-dom';
+import { t } from '@lingui/macro';
+import { FormGroup, Tooltip } from '@patternfly/react-core';
+import { QuestionCircleIcon as PFQuestionCircleIcon } from '@patternfly/react-icons';
+import styled from 'styled-components';
+
+import { CredentialsAPI, CredentialTypesAPI } from '@api';
+import Lookup from '@components/Lookup';
+
+const QuestionCircleIcon = styled(PFQuestionCircleIcon)`
+  margin-left: 10px;
+`;
+
+class CredentialsLookup extends React.Component {
+  constructor(props) {
+    super(props)
+
+    this.state = {
+      credentials: props.credentials,
+      selectedCredentialType: {label: "Machine", id: 1, kind: 'ssh'},
+      credentialTypes: []
+    }
+
+    this.handleCredentialTypeSelect = this.handleCredentialTypeSelect.bind(this);
+    this.loadCredentials = this.loadCredentials.bind(this);
+    this.loadCredentialTypes = this.loadCredentialTypes.bind(this);
+    this.toggleCredential = this.toggleCredential.bind(this);
+  }
+
+  componentDidMount() {
+    this.loadCredentials({page: 1, page_size: 5, order_by:'name'})
+    this.loadCredentialTypes();
+  }
+
+  componentDidUpdate(prevState) {
+    const {selectedType} = this.state
+    if (prevState.selectedType !== selectedType) {
+      Promise.all([this.loadCredentials()]);
+    }
+  }
+
+  async loadCredentialTypes() {
+    const {onError} =this.props
+    try {
+      const { data } = await CredentialTypesAPI.read()
+          const acceptableTypes = ["machine", "cloud", "net", "ssh", "vault"]
+          const credentialTypes =[]
+          data.results
+            .forEach(cred => {
+              acceptableTypes.forEach(aT => {
+                if (aT === cred.kind) {
+                  // This object has several repeated values as some of it's children
+                  // require different field values.
+                  cred = {
+                    id: cred.id, kind: cred.kind, type: cred.namespace,
+                    value: cred.name, label: cred.name, isDisabled: false
```

Comment:
Hopefully when I refactor `Lookup` some of the duplicated fields can be removed.